tests/qnodes/test_qnode_decorator.py

```diff
@@ -152,9 +154,26 @@ def circuit(a):
 
     assert isinstance(circuit, ReversibleQNode)
 
+
+def test_reversible_diff_method_exception(monkeypatch):
+    """Test that an exception is raised if the reversible diff_method
+    is specified for a device which does not have reversible capability."""
+    dev = qml.device("default.qubit", wires=1)
+    capabilities = {**dev._capabilities}
+    capabilities["reversible"] = False
+    monkeypatch.setattr(dev, "_capabilities", capabilities)
+
+    with pytest.raises(ValueError, match="Reversible differentiation method not supported"):
+
+        @qnode(dev, diff_method="reversible")
+        def circuit(a):
+            qml.RX(a, wires=0)
+            return qml.expval(qml.PauliZ(wires=0))
```

Comment:
Nitpick to make it clearer which part of the logic the error comes from:

```suggestion
    def circuit(a):
        qml.RX(a, wires=0)
        return qml.expval(qml.PauliZ(wires=0))
            
    with pytest.raises(ValueError, match="Reversible differentiation method not supported"):
        qml.QNode(circuit, dev, diff_method="reversible")
```

Reply:
Oops. I see below that there are lots of tests using the existing style, which I potentially wrote 😆

Reply:
Note that this test is in the file `test_qnode_decorator`, so I feel it's appropriate to put the decorator inside the context

Reply:
I used that other pattern in the qnode-specific test file